draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,669 @@
+- **Sign-in experience settings**: The configuration settings that define the behavior of the sign-in experience。 Including the required identifiers, verification methods, profile data, and MFA settings. Developers can customize the sign-in experience settings to meet their specific requirements.
+- **Experience API**: A set of APIs that allow end-users to interact with the Logto for identity verification and profile fulfillment.
+- **Experience app**: The front-end application that provides the user interface for sign-in experience.
+- **Interaction**: A short-lived session that starts when a user initiates the sign-in experience. By default, it has an expiration time of one hour.
+- **Social identity**: The data used to identify a user within a social identity provider, such as Google.
+- **SSO (Single sign-on) identity**: The data used to identify a user within an identity provider that is configured as an enterprise SSO connector in Logto.
+- **Identifier**: A piece of information used to identify a user in Logto. It could be a username, email, phone number, social identity, SSO identity, among others. Each identifier should be unique within its type.
+- **Verification code**: A one-time code sent to the user's email or phone number for identity verification.
+- **Verification record**: A group of data can be used to verify a user's identity by such as password, verification code, social identity, etc.
+- **Verification ID**: A unique identifier for a verification record. It's a random string generated by the Logto server when a verification record is initiated. Users need to provide the verification ID along with additional security information to verify their identity. The verification ID can also be used to track the status of the verification record within the interaction.
+- **Multi-factor authentication (MFA)**: A security process that requires more than one verification factor to complete the interaction. It can be used as an additional security check during the sign-in experience.
+- **First party application**: A application that is built and managed by the same company that uses Logto for user authentication.
+- **Third party application**: A application that is owned and managed by a different company that uses Logto as a external identity provider. E.g. A external application that uses Logto as a enterprise SSO provider.
+
+### 3.2 User experience
+
+The user sign-in experience in Logto consists of the following key steps:
+
+1. **Sign-in experience initiation**: When the client application sends an authentication request to Logto, the user is redirected to Logto's experience app, initiating a new interaction for the authentication process.
+2. **Identification**: The user must provide an verified identifier to Logto, such as a username, email address, phone number, social identity, or SSO identity. This identifier is used to recognize the user during the interaction.
```

Comment:
We could update the identifier section once the passkey as a primary verification method is supported. Since we put a `such as` as a prefix, I think is fine to just list a few standard ones here, and we may extend whatever extra identifiers we need in the future.